### ISSUE TITLE
Fix false positives for #{&} in no-descending-specificity

### ISSUE THREAD

**Issue by primetwig** [stylelint/stylelint#3417]
> Clearly describe the bug

There are false positives for rule `no-descending-specificity` when some manipulations are applied to a selector.

> Which rule, if any, is the bug related to?

`no-descending-specificity`

> What CSS is needed to reproduce the bug?

e.g.

```css
.foo {
    &--a,
    &--a#{&}--b {
      div {
        padding: 10px;
      }
    }
}
```

> What stylelint configuration is needed to reproduce the bug?

default

> Which version of stylelint are you using?

`9.2.0`

> How are you running stylelint: CLI, PostCSS plugin, Node API?

"CLI with `stylelint '**/*.{css,scss}'`"

> Does the bug relate to non-standard syntax (e.g. SCSS, Less etc.)?

"Yes, it's related to SCSS injections into selector string."

> What did you expect to happen?

"No warnings to be flagged."

> What actually happened (e.g. what warnings or errors did you get)?

"The following warnings were flagged:"

```shell
file.scss
4:6    Expected selector ".foo--a#{.foo}--b div" to come before selector ".foo--a div" (no-descending-specificity)
```

**Comment by jeddy3** [comment]
@primetwig Thanks for the report and for using the template.

As the issue has been labeled "help wanted", please consider [contributing a fix](https://github.com/stylelint/stylelint/blob/master/docs/developer-guide/rules.md#fixing-a-bug-in-an-existing-rule).

The fix might be as simple as using the [`isStandardSyntaxSelector`](https://github.com/stylelint/stylelint/blob/master/lib/utils/isStandardSyntaxSelector.js) util [before this line](https://github.com/stylelint/stylelint/blob/master/lib/rules/no-descending-specificity/index.js#L59), and adding a test specific to scss syntax ([like so](https://github.com/stylelint/stylelint/blob/master/lib/rules/font-weight-notation/__tests__/index.js#L307-L322)).

**Comment by primetwig** [comment]
@jeddy3, thank you for your response.
Label has been added by someone but me.
I'm not sure I can make a contribution (I'm not familiar with this codebase).

**Comment by hudochenkov** [comment]
Done in https://github.com/stylelint/stylelint/pull/3420
